rust/connlib/clients/apple/src/lib.rs

```diff
@@ -137,24 +138,22 @@ impl Callbacks for CallbackHandler {
                 return;
             }
         };
-
-        self.inner.on_set_interface_config(
-            tunnel_address_v4.to_string(),
-            tunnel_address_v6.to_string(),
-            dns_addresses,
-        );
-    }
-
-    fn on_update_routes(&self, route_list_4: Vec<Ipv4Network>, route_list_6: Vec<Ipv6Network>) {
         match (
-            serde_json::to_string(&V4RouteList::new(route_list_4)),
-            serde_json::to_string(&V6RouteList::new(route_list_6)),
+            serde_json::to_string(&dns_addresses),
```

Comment:
There was a bug here, this introduces double serialization of `dns_addresses` since it's serialized just above.

Reply:
whoops!